python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py

```diff
@@ -132,42 +135,46 @@ def sync_and_poll(
-                )
-
-                for line in log_lines[logged_lines:]:
-                    sys.stdout.write(line + "\n")
-                    sys.stdout.flush()
-                logged_lines = len(log_lines)
-
-                # if there's a next attempt, this one will have no more log messages
-                if logged_attempts < cur_attempt - 1:
-                    logged_lines = 0
-                    logged_attempts += 1
-
-            state = job_details.get("job", {}).get("status")
-
-            if state in (AirbyteState.RUNNING, AirbyteState.PENDING, AirbyteState.INCOMPLETE):
-                continue
-            elif state == AirbyteState.SUCCEEDED:
-                break
-            elif state == AirbyteState.ERROR:
-                raise Failure(f"Job failed: {job_id}")
-            elif state == AirbyteState.CANCELLED:
-                raise Failure(f"Job was cancelled: {job_id}")
-            else:
-                raise Failure(f"Encountered unexpected state `{state}` for job_id {job_id}")
+        try:
+            while True:
+                if poll_timeout and start + poll_timeout < time.monotonic():
+                    raise Failure(
+                        f"Timeout: Airbyte job {job_id} is not ready after the timeout {poll_timeout} seconds"
+                    )
+                time.sleep(poll_interval)
+                job_details = self.get_job_status(job_id)
+                cur_attempt = len(job_details.get("attempts", []))
+                # spit out the available Airbyte log info
+                if cur_attempt:
+                    log_lines = (
+                        job_details["attempts"][logged_attempts].get("logs", {}).get("logLines", [])
+                    )
+
+                    for line in log_lines[logged_lines:]:
+                        sys.stdout.write(line + "\n")
+                        sys.stdout.flush()
+                    logged_lines = len(log_lines)
+
+                    # if there's a next attempt, this one will have no more log messages
+                    if logged_attempts < cur_attempt - 1:
+                        logged_lines = 0
+                        logged_attempts += 1
+
+                state = job_details.get("job", {}).get("status")
+
+                if state in (AirbyteState.RUNNING, AirbyteState.PENDING, AirbyteState.INCOMPLETE):
+                    continue
+                elif state == AirbyteState.SUCCEEDED:
+                    break
+                elif state == AirbyteState.ERROR:
+                    raise Failure(f"Job failed: {job_id}")
+                elif state == AirbyteState.CANCELLED:
+                    raise Failure(f"Job was cancelled: {job_id}")
+                else:
+                    raise Failure(f"Encountered unexpected state `{state}` for job_id {job_id}")
+        finally:
+            # make sure that the Airbyte job does not outlive the python process
```

Comment:
probably worth explicitly saying this part in the comment: "Sending a cancel request on an airbyte job that succeeded does not have any effect, so it's ok to send it regardless of the state of the job."

Or putting this within an except: instead? It may be ok, but is there a reason not to put it in an except, if only for clarity?